File: palindrome/palindrome.c
// check given number is palindrome or not

#include <stdio.h>

int main() {
  int reversed = 0;
  int n;
  int originalNumber;
  int remainder;

  printf("Enter a number: ");
  scanf("%d", &n);

  originalNumber = n;

  while(originalNumber != 0) {
    remainder = originalNumber % 10;
    reversed = reversed * 10 + remainder;
    originalNumber /= 10;
  }

  if(reversed == n) {
    printf("%d is palindrome\n", n);
  } else {
    printf("%d is not palindrome\n", n);
  }
}

File: palindrome/palindrome.js
let number = parseInt(process.argv[2]);
let originalNumber;
let reserved = 0;
let remainder;

originalNumber = number;

while (originalNumber != 0) {
  remainder = parseInt(originalNumber % 10);
  reserved = reserved * 10 + remainder;
  originalNumber = parseInt(originalNumber / 10);
}

if (reserved == number) {
  console.log(`${number} is palindrome`);
} else {
  console.log(`${number} is not palindrome`);
}
